array_api_compat/common/_helpers.py

```diff
@@ -91,7 +92,7 @@ def is_cupy_array(x):
     import cupy as cp
 
     # TODO: Should we reject ndarray subclasses?
-    return isinstance(x, (cp.ndarray, cp.generic))
+    return isinstance(x, cp.ndarray)
```

Comment:
`cp.generic is np.generic`. So pure-numpy use cases would take a cupy-specific code path.
Need to move this to a separate PR.

Reply:
Blocked by https://github.com/data-apis/array-api-compat/pull/207

Reply:
gh-207 is merged, so can rebase on main

Reply:
already done